app/src/views/features/sessions/SessionViewer/sessionEventsUtils.ts

```diff
@@ -30,3 +34,64 @@ export const filterOutLargeNetworkResponses = (
     }
   });
 };
+
+const isConsoleLogEvent = (rrwebEvent: RRWebEventData): boolean => {
+  if (rrwebEvent.type === EventType.IncrementalSnapshot) {
+    //@ts-ignore
+    return rrwebEvent.data.source === IncrementalSource.Log;
+  }
+  if (rrwebEvent.type === EventType.Plugin) {
+    return rrwebEvent.data.plugin === "rrweb/console@1";
+  }
+
+  return false;
+};
+
+export const getConsoleLogs = (
+  rrwebEvents: RRWebEventData[],
+  startTime: number
+): ConsoleLog[] => {
+  return rrwebEvents
+    .map((event) => {
+      let logData: LogData = null;
+      if (isConsoleLogEvent(event)) {
+        if (event.type === EventType.IncrementalSnapshot) {
+          logData = (event.data as unknown) as LogData;
+        } else if (event.type === EventType.Plugin) {
+          logData = event.data.payload as LogData;
+        }
+      }
+
+      return (
+        logData && {
+          ...logData,
+          timeOffset: Math.floor((event.timestamp - startTime) / 1000),
+        }
+      );
+    })
+    .filter((event) => !!event);
+};
+
+export const filterOutConsoleLogs = (
+  rrwebEvents: RRWebEventData[]
+): RRWebEventData[] => {
+  return rrwebEvents.filter((event) => !isConsoleLogEvent(event));
+};
+
+export const filterOutSessionEvents = (
+  sessionEvents: RQSessionEvents,
+  includedDebugInfo: IncludedDebugInfo
+): RQSessionEvents => {
+  let filteredSessionEvents = cloneDeep(sessionEvents);
```

Comment:
You could avoid `cloneDeep` here, as it is an expensive operation.
`filterOutConsoleLogs(events) + network events`